docs/rules/no-obscure-range.md

```diff
@@ -0,0 +1,48 @@
+---
+pageClass: "rule-details"
+sidebarDepth: 0
+title: "regexp/no-obscure-range"
+description: "disallow obscure character ranges"
+---
+# regexp/no-obscure-range
+
+> disallow obscure character ranges
+
+- :exclamation: <badge text="This rule has not been released yet." vertical="middle" type="error"> ***This rule has not been released yet.*** </badge>
+- :gear: This rule is included in `"plugin:regexp/recommended"`.
+
+## :book: Rule Details
+
+The character range operator (the `-` inside character classes) can easily be misused (mostly unintentionally) to construct non-obvious character class. This rule will disallow all non-obvious uses of the character range operator.
+
+<eslint-code-block>
+
+```js
+/* eslint regexp/no-obscure-range: "error" */
+
+/* ✓ GOOD */
+var foo = /[a-z]/;
+var foo = /[J-O]/;
+var foo = /[1-9]/;
+var foo = /[\x00-\x40]/;
+var foo = /[\0-\uFFFF]/;
+var foo = /[\0-\u{10FFFF}]/u;
+var foo = /[\1-\5]/;
+var foo = /[\cA-\cZ]/;
+
+/* ✗ BAD */
+var foo = /[A-\x43]/;
+var foo = /[\41-\x45]/;
+var foo = /[*/+-^&|]/;
+```
+
+</eslint-code-block>
+
+## :wrench: Options
+
+Nothing.
```

Comment:
What do you think about adding the option to add a character range that your team is familiar with?
For example, I clearly understand that `ｱ-ﾝ`(\uFF71-\uFF9D) is in order because I use Japanese. In addition to these, `①-⑳` and `⑴-⒇` have order codes.

`prefer-range` rule has slightly similar options.
https://ota-meshi.github.io/eslint-plugin-regexp/rules/prefer-range.html#options

Reply:
> I clearly understand that ｱ-ﾝ(\uFF71-\uFF9D) is in order because I use Japanese

As somebody that recently started learning Japanese, let me say that the order in the [Kana script](https://www.compart.com/en/unicode/scripts/Kana) isn't obvious to me at all...

But I agree that we should allow our users to specify additional allowed character ranges. There are a lot of different scripts where ranges make sense (even if they don't make sense to me),

> `prefer-range` rule has slightly similar options.

Maybe we should use a [setting](https://eslint.org/docs/user-guide/configuring/configuration-files#adding-shared-settings)? Both `prefer-range` and `no-obscure-range` have to know what ranges are allowed/wanted by the user.